ethereum/eth/src/test/java/tech/pegasys/pantheon/ethereum/eth/messages/LimitedTransactionsMessagesTest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.eth.messages;
+
+import static org.junit.Assert.assertEquals;
+
+import tech.pegasys.pantheon.ethereum.core.BlockDataGenerator;
+import tech.pegasys.pantheon.ethereum.core.Transaction;
+
+import java.util.Set;
+
+import org.junit.Test;
+
+public class LimitedTransactionsMessagesTest {
+
+  private final BlockDataGenerator generator = new BlockDataGenerator();
+  private final Set<Transaction> sampleTxs = generator.transactions(1);
+  private final TransactionsMessage sampleTransactionMessages =
+      TransactionsMessage.create(sampleTxs);
+  private final LimitedTransactionsMessages sampleLimitedTransactionsMessages =
+      new LimitedTransactionsMessages(sampleTransactionMessages, sampleTxs);
+
+  @Test
+  public void createLimited() {
+    final Set<Transaction> txs = generator.transactions(6000);
+    final LimitedTransactionsMessages firstMessage = LimitedTransactionsMessages.createLimited(txs);
+    assertEquals(5219, firstMessage.getIncludedTransactions().size());
```

Comment:
I think its worth summing up the final size (in bytes) of all these messages and asserting the total size is < 2 x limit

Reply:
Done.